janusgraph-benchmark/src/main/java/org/janusgraph/MultiDropBenchmark.java

```diff
@@ -0,0 +1,146 @@
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.ADJUST_LIMIT;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.HARD_MAX_LIMIT;
+
+@BenchmarkMode(Mode.AverageTime)
+@Fork(1)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class MultiDropBenchmark {
+
+    @Param({"5000"})
+    int verticesAmount;
+
+    @Param({"true", "false"})
+    boolean isMultiDrop;
+
+    JanusGraph graph;
+
+    public WriteConfiguration getConfiguration() {
+        ModifiableConfiguration config = GraphDatabaseConfiguration.buildGraphConfiguration();
+        config.set(GraphDatabaseConfiguration.STORAGE_BACKEND,"cql");
+        config.set(CQLConfigOptions.LOCAL_DATACENTER, "dc1");
+        config.set(GraphDatabaseConfiguration.USE_MULTIQUERY, true);
+        return config.getConfiguration();
+    }
+
+    @Setup
+    public void setUp() throws Exception {
+        graph = JanusGraphFactory.open(getConfiguration());
+
+        boolean alreadyAdded = true;
+        JanusGraphManagement mgmt = graph.openManagement();
+        if (!mgmt.containsPropertyKey("id")) {
+            mgmt.makePropertyKey("id").dataType(Integer.class).cardinality(Cardinality.SINGLE).make();
+            PropertyKey nameProp = mgmt.makePropertyKey("name").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+            mgmt.makePropertyKey("details").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+            mgmt.buildIndex("nameIndex", Vertex.class).addKey(nameProp).buildCompositeIndex();
+
+            alreadyAdded = false;
+        }
+
+        mgmt.commit();
+
+        if (!alreadyAdded) {
+            addVertices();
+        }
+    }
+
+    @TearDown
+    public void tearDown() {
+        graph.close();
+    }
+
+    @Benchmark
+    public Integer dropVertices() {
+
+        JanusGraphTransaction tx = graph.buildTransaction().start();
+        Integer dropCount = 0;
+        if (isMultiDrop) {
+
+            List<JanusGraphVertex> vertices = tx.traversal()
+                .V()
+                .has("name", "name_test")
+                .toList()
+                .stream()
+                .map(v -> (JanusGraphVertex) v)
+                .collect(Collectors.toList());
+
+            dropCount = tx.multiQuery(vertices).drop();
+        } else {
+            dropCount = tx.traversal()
+                .V()
+                .has("name", "name_test")
+                .drop()
+                .toList()
+                .size();
+        }
+
+        tx.rollback();
+
+        /*if (dropCount != verticesAmount) {
+            throw new AssertionError("Only " + dropCount + " vertices were dropped, expected " + verticesAmount);
+        }*/
```

Comment:
Seems this is better suited for unit testing than benchmark testing.

Reply:
Will add unit test to validate deletion

Reply:
Done